api/src/test/java/com/epam/pipeline/acl/docker/DockerRegistryApiServiceTest.java

```diff
@@ -0,0 +1,363 @@
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldDenyGetCertificateContentWhenPermissionIsNotGranted() {
+        initAclEntity(dockerRegistry);
+
+        assertThrows(AccessDeniedException.class, () -> dockerRegistryApiService.getCertificateContent(ID));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldGetConfigScript() {
+        doReturn(BYTE_RESULT).when(mockDockerRegistryManager).getConfigScript(ID);
+
+        assertThat(dockerRegistryApiService.getConfigScript(ID)).isEqualTo(BYTE_RESULT);
+    }
+
+    @Test
+    @WithMockUser(SIMPLE_USER)
+    public void shouldGetConfigScriptWhenPermissionIsGranted() {
+        doReturn(BYTE_RESULT).when(mockDockerRegistryManager).getConfigScript(ID);
+        initAclEntity(dockerRegistry, AclPermission.READ);
+
+        assertThat(dockerRegistryApiService.getConfigScript(ID)).isEqualTo(BYTE_RESULT);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldDenyGetConfigScriptWhenPermissionIsNotGranted() {
+        initAclEntity(dockerRegistry);
+
+        assertThrows(AccessDeniedException.class, () -> dockerRegistryApiService.getConfigScript(ID));
+    }
+
+    private void initDockerRegistryAclTree() {
+        toolGroup.setTools(toolList);
+        toolGroupWithoutPermission.setTools(toolList);
+        dockerRegistryWithTools.setGroups(toolGroups);
+        initAclEntity(toolRead, AclPermission.READ);
+        initAclEntity(toolWithoutPermission);
+        initAclEntity(toolGroup, AclPermission.READ);
+        initAclEntity(toolGroupWithoutPermission);
+        initAclEntity(emptyToolGroupWithoutPermission, AclPermission.READ);
+    }
+
+    private void assertPartialDockerRegistryAclTree(final DockerRegistry registry) {
+        assertDockerRegistryAclHierarchy(registry);
+        final List<? extends AbstractSecuredEntity> toolGroupWithoutPermissionLeaves =
+                registry.getChildren().get(1).getLeaves();
+        assertThat(toolGroupWithoutPermissionLeaves.size()).isEqualTo(1);
+        assertThat(toolGroupWithoutPermissionLeaves.get(0)).isEqualTo(toolRead);
+    }
+
+    private void assertWholeDockerRegistryAclTree(final DockerRegistry registry) {
+        assertDockerRegistryAclHierarchy(registry);
+        final List<? extends AbstractSecuredEntity> toolGroupWithoutPermissionLeaves =
+                registry.getChildren().get(1).getLeaves();
+
+        assertThat(toolGroupWithoutPermissionLeaves.size()).isEqualTo(toolList.size());
+        assertThat(toolGroupWithoutPermissionLeaves).isEqualTo(toolList);
+    }
+
+    private void assertDockerRegistryAclHierarchy(final DockerRegistry registry) {
+        final List<AbstractHierarchicalEntity> dockerRegistryChildren = registry.getChildren();
+        final List<? extends AbstractSecuredEntity> toolGroupLeaves = registry.getChildren().get(0).getLeaves();
```

Comment:
Could we refactor this logic somehow: list.get(index) is not clear? (Maybe pick methods for each requested entity? by Id?)